sql/binlog.cc

```diff
@@ -9422,6 +9429,28 @@ TC_LOG::enum_result MYSQL_BIN_LOG::commit(THD *thd, bool all)
       const ulong timeout= thd->variables.lock_wait_timeout;
 
       DBUG_PRINT("debug", ("Acquiring binlog protection lock"));
+
+#ifdef HAVE_REPLICATION
+      DBUG_EXECUTE_IF("delay_slave_worker_0", {
+        if (has_commit_order_manager(thd))
+        {
+          Slave_worker *worker= dynamic_cast<Slave_worker *>(thd->rli_slave);
+
+          static bool skip_first_query= true;
```

Comment:
I am not sue if it really works when test is run second time without server restart.

Reply:
I improved this a little bit and verified that it works fine with `--repeat` with and without deadlock.